src/idpi/operators/vertical_reduction.py

```diff
@@ -297,10 +297,12 @@ def integrate_k(field, operator, mode, height, h_bounds, hsurf=None):
     # ... note that the dimension "generalVericalLayer" is lost of this condition is
     # nowhere satisfied
     field_in_h_bounds = field_on_fl.where((hfl >= h_bottom) & (hfl <= h_top)).dropna(
-        "generalVerticalLayer"
+        dim="generalVerticalLayer",
+        how="all",
```

Comment:
this is a significant change in behaviour, and the tests were insensitive to it (since there are no changes to testing). Could we test this expected behaviour (i.e. "all" vs "any")?

Reply:
Sure. If a single column is undefined, then the entire output is empty. Should I add this as a test case?

Reply:
I do not know. I think the proposed change here is correct since we do not want to drop entire column because there is any nan, I just do not understand how this was working before and passing tests. But also it seems that this algorithm is reducing the horizontal domain of the resulting field, instead of returning a field with values nan in horizontal grid points where the hfl within [h_bottom, h_top] condition is not satisfied. So I do not understand why we need a dropna here, particularly when there is a `fillna(0.0)` later in the algorithm. we could briefly discuss it to understand it, but anyhow I think it does not belong to this PR

Reply:
I believe the intention is drop the levels that do not intersect with h bounds in order to reduce the number summations of 0 values. I have had to change the `fill(0.0)` line in the next as well because it did not agree with the way the fieldextra is handling some of these undefined values.

Reply:
note to self: add comment here